homework/Program.cs

```diff
@@ -10,6 +8,215 @@ internal class Program
+
+            PrintArray(baseArray);
+
+            Console.WriteLine();
+            Console.WriteLine("Array filtered:");
+
+            //  Runs over the baseArray and copy only the elements which are not allocated in tempFilteredArray
+            int size = 0;
+
+            for (int index = 0; index < baseArray.Length; index++)
+            {
+                // checks if the value is stored in tempfilteredarray
+                if(Array.IndexOf(tempfilteredArray, baseArray[index]) == -1)
+                {
+                    // copy the value to the temparray
+                    tempfilteredArray[size] = baseArray[index];
+                    size++;
+                }
+            }
+            
+            //  size would be the size of the pre filtered array, now, only we need to do is to crop all the 0's
+            int[] filteredArray = new int[size];
+
+            for (int index = 0; index < size; index++)
+            {
+                filteredArray[index] = tempfilteredArray[index];
+            }
+
+            //  Writes in Console the array filtered
+            PrintArray(filteredArray);
+        }
+
+        static void PrintArray(int[] array)
+        {
+            foreach (var item in array)
+            {
+                Console.Write(item.ToString());
+            }
+        }
+
+        static void discoverJustAA(int start, int finish)
+        {
+            string hasDoubleAA = "";
+
+            // Checks every number from A to B and prints every number that has double AA
+            for (int currentNumber = start; currentNumber < finish + 1; currentNumber++)
+            {
+                int count = 0;
+                foreach (var c in convertToDuoDecimal(currentNumber))
+                {
+                    if (c == 'A')
+                    {
+                        count++;
+                    }
+                }
+                if (count == 2)
+                {
+                    hasDoubleAA += currentNumber.ToString() + " ";
+                }
+            }
+
+            if (hasDoubleAA.Length != 0)
+            {
+                Console.WriteLine(hasDoubleAA);
+            } else
+            {
+                Console.WriteLine("None");
+            }
+        }
+
+        static void discoverJustAA(int number)
+        {
+            // The count that checks if only 2 AA's appears in the number
+            int count = 0;
+
+            // if there's only a number, check if has double AA
+            foreach (var c in convertToDuoDecimal(number))
+            {
+                if (c == 'A')
+                {
+                    count++;
+                }
+            }
+            if (count == 2)
+            {
+                Console.WriteLine(number.ToString());
+            } else
+            {
+                Console.WriteLine("None");
+            }
+        }
+
+        static char[] convertToDuoDecimal(int number)
+        {
+            string duodecimalNumber = "";
```

Comment:
duoNumber variable is not necessary, just return string and iterate in in foreach (line 171)